[Swift in WebKit] The Swift implementation of WKGroupSession is incorrect

rr-codes opened the pull request:
#### 87ffa9eb985d4367ea42c6c953d37b077404a7eb
<pre>
[Swift in WebKit] The Swift implementation of WKGroupSession is incorrect
<a href="https://bugs.webkit.org/show_bug.cgi?id=292830">https://bugs.webkit.org/show_bug.cgi?id=292830</a>
<a href="https://rdar.apple.com/151089048">rdar://151089048</a>

Reviewed by Elliott Williams.

Fix the implementation of WKGroupSession to properly use `@_objcImplementation.`

* Source/WebKit/Configurations/WebKitSwift.xcconfig:

- Remove WK_EXCLUDED_COORDINATOR_FILES and WK_HAVE_COORDINATOR in favor of just using
normal compile-time guards. Plus, neither of these were even doing anything to begin
with since `WKGroupSession.swift` never existed prior to this PR.

- Remove `WK_EXCLUDED_TEXT_STYLE_MANAGER_FILES`, which doesn&apos;t actually exist.

- Add missing semicolons for consistent style

* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesCoordinator.h:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesCoordinator.mm:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSession.h:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSession.mm:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSessionNotifier.h:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSessionNotifier.mm:
* Source/WebKit/UIProcess/WebPageProxy.cpp:

- Enforce consistency of the compilation conditions between the different files by adding
`USE(APPLE_INTERNAL_SDK)` in places it was previously missing from.

* Source/WebKit/WebKitSwift/GroupActivities/WKGroupSession.h: Renamed from Source/WebKit/UIProcess/Cocoa/GroupActivities/WKGroupSession.h.

- Move to WKS directory for consistency
- Remove unneeded forward declarations
- Use `nullable` istead of `_Nullable` for consistency
- Remove unnecessary `init` declaration
- Remove unnecessary `strong` attributes

* Source/WebKit/WebKitSwift/GroupActivities/WKGroupSession.swift: Renamed from Source/WebKit/WebKitSwift/GroupActivities/GroupSession.swift.

- Repace `canImport` with the preferred way, using normal compilation conditions.
- Alphabetize the imports
- Use `@_objcImplementation`, and remove unnecessary `public`, `@objc`, and availability attributes
- Enforce consistency by renaming `URLActivityWrapper` to `WKURLActivity`
- Enforce consistency by creating an adopter initializer for `WKGroupSessionState`
- Enforce consistency by removing `self.` when applicable

Canonical link: <a href="https://commits.webkit.org/294828@main">https://commits.webkit.org/294828@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ca8ed1af0056e56aee010d574596ba28fe2f63aa

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/103255 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/22931 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/13251 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/108425 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/53898 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/23268 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/31442 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/78477 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/53898 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/106261 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/18037 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/93147 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/58810 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/17878 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/53253 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/87684 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/11252 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/110801 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/30393 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/22409 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/87480 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/30758 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/89346 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/87114 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22158 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/31963 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/9678 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/24716 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/30322 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/35641 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/30128 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/33455 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/31690 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -917,7 +917,7 @@ WebPageProxy::WebPageProxy(PageClient& pageClient, WebProcessProxy& process, Ref
 #endif
 
-#if ENABLE(MEDIA_SESSION_COORDINATOR) && HAVE(GROUP_ACTIVITIES)
+#if USE(APPLE_INTERNAL_SDK) && ENABLE(MEDIA_SESSION_COORDINATOR) && HAVE(GROUP_ACTIVITIES)
```
Can you help me understand why this is needed? GroupActivities is a public framework, and I'd expect the SDK check to be written inside Platform.h instead (here, probably in the definition of `ENABLE_MEDIA_SESSION_COORDINATOR`).

emw-apple reviewed (approved):
We talked offline and I understand the need for the USE_APPLE_INTERNAL_SDK checks now. Please leave a FIXME to indicate that we can remove these once we have an SPI module for GroupActivities checked in.

rr-codes committed:
[Swift in WebKit] The Swift implementation of WKGroupSession is incorrect
https://bugs.webkit.org/show_bug.cgi?id=292830
rdar://151089048

Reviewed by Elliott Williams.

Fix the implementation of WKGroupSession to properly use `@_objcImplementation.`

* Source/WebKit/Configurations/WebKitSwift.xcconfig:

- Remove WK_EXCLUDED_COORDINATOR_FILES and WK_HAVE_COORDINATOR in favor of just using
normal compile-time guards. Plus, neither of these were even doing anything to begin
with since `WKGroupSession.swift` never existed prior to this PR.

- Remove `WK_EXCLUDED_TEXT_STYLE_MANAGER_FILES`, which doesn't actually exist.

- Add missing semicolons for consistent style

* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesCoordinator.h:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesCoordinator.mm:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSession.h:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSession.mm:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSessionNotifier.h:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSessionNotifier.mm:
* Source/WebKit/UIProcess/WebPageProxy.cpp:

- Enforce consistency of the compilation conditions between the different files by adding
`USE(APPLE_INTERNAL_SDK)` in places it was previously missing from.

* Source/WebKit/WebKitSwift/GroupActivities/WKGroupSession.h: Renamed from Source/WebKit/UIProcess/Cocoa/GroupActivities/WKGroupSession.h.

- Move to WKS directory for consistency
- Remove unneeded forward declarations
- Use `nullable` istead of `_Nullable` for consistency
- Remove unnecessary `init` declaration
- Remove unnecessary `strong` attributes

* Source/WebKit/WebKitSwift/GroupActivities/WKGroupSession.swift: Renamed from Source/WebKit/WebKitSwift/GroupActivities/GroupSession.swift.

- Repace `canImport` with the preferred way, using normal compilation conditions.
- Alphabetize the imports
- Use `@_objcImplementation`, and remove unnecessary `public`, `@objc`, and availability attributes
- Enforce consistency by renaming `URLActivityWrapper` to `WKURLActivity`
- Enforce consistency by creating an adopter initializer for `WKGroupSessionState`
- Enforce consistency by removing `self.` when applicable

Canonical link: https://commits.webkit.org/294828@main
